### ISSUE TITLE
[Bug]: Linter executables checked at startup?

### ISSUE THREAD

**Issue by LandonSchropp** [LunarVim/LunarVim#1749]
### Problem description

**TL/DR: LunarVim should only check if the linter executables are present when opening the relevant file types.**

I use eslint_d to lint my JavaScript projects. Typically, I install `eslint` and `eslint_d` locally in the `node_modules` folder and include `node_modules/.bin` in my path. This allows me to run `eslint_d` in each project, and use slightly different versions and configurations.

It looks like LunarVim checks for the `eslint_d` executable on startup. However, since this is always locally installed, I get an error on non-JavaScript projects. I don't think it makes sense for LunarVim to be checking this.

Here's my config:

``` lua
lvim.lang.javascript.formatters = { { exe = "eslint_d" } }
lvim.lang.javascriptreact.formatters = { { exe = "eslint_d" } }
lvim.lang.javascript.linters = { { exe = "eslint_d" } }
lvim.lang.javascriptreact.linters = { { exe = "eslint_d" } }
```

### LunarVim version

Latest rolling

### Neovim version (>= 0.5)

v0.5.0

### Relevant log output

```shell
Here's the error I get from LunarVim:


[lunarvim] [WARN  21:56:33] .../landon/.local/share/lunarvim/lvim/lua/lvim/core/log.lua:10: Not found: eslint_d
```

### Screenshots

Here's the warning. It's a bit annoying.

<img width="1534" alt="Screen Shot 2021-10-11 at 9 58 42 PM" src="https://user-images.githubusercontent.com/361591/136883818-5a41ab58-6784-45e1-8666-4a8ede89a59b.png">

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by LandonSchropp** [comment]
@kylo252 I posted about this in Discord, but it seems like that style of config is now causing a different issue. When I try to save a JSX file now, I get the following prompt:

<img width="296" alt="Screen Shot 2021-10-12 at 1 14 58 PM" src="https://user-images.githubusercontent.com/361591/137010796-3e104912-53bb-45b2-815a-15f72623b8a9.png">

**Comment by LandonSchropp** [comment]
@kylo252 Sorry, my bad! I missed the part about disable the language server formatters.
